pulumi/index.ts

```diff
@@ -0,0 +1,97 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import * as awsx from "@pulumi/awsx";
+
+import * as fs from 'fs';
+import * as YAML from 'yaml'
+
+const repo = new awsx.ecr.Repository("tezos-k8s");
+
+// Manual step: create a pulumi_values.yaml in the top level dir
+// with mkchain command:
+//   mkchain pulumi
+//
+
+const chainName = process.env.CHAIN_NAME || "pulumi";
+
+const helmValuesFile = fs.readFileSync(chainName + '_values.yaml', 'utf8')
+const helmValues = YAML.parse(helmValuesFile)
+
+const nginxIngressHelmValues_file = fs.readFileSync('nginx_ingress_values.yaml', 'utf8')
+const nginxIngressHelmValues = YAML.parse(nginxIngressHelmValues_file)
+
+let images : {};
+images = helmValues['tezos_k8s_images'] || {};
+
+const imagelist = ["baker_endorser", "chain_initiator", "config_generator",
+	           "rpc_auth", "wait_for_bootstrap", "zerotier"]
+
+for (let image of imagelist) {
+    images[image] = repo.buildAndPushImage("../" + image.replace(/_/g, "-"))
+}
+
+helmValues["tezos_k8s_images"] = images;
```

Comment:
I don't know where my previous comment here went, but this can be improved to:
```suggestion
const defaultHelmValuesFile = fs.readFileSync("../charts/tezos/values.yaml", 'utf8')
const defaultHelmValues = YAML.parse(defaultHelmValuesFile)

const helmValuesFile = fs.readFileSync(chainName + '_values.yaml', 'utf8')
const helmValues = YAML.parse(helmValuesFile)

const tezosK8sImages = defaultHelmValues["tezos_k8s_images"]
const pulumiTaggedImages = Object.entries(tezosK8sImages).reduce(
  (obj, [imageKey]) => {
    obj[imageKey] = repo.buildAndPushImage(`../${imageKey.replace(/_/g, "-")}`)
    return obj
  },
  {}
)

helmValues["tezos_k8s_images"] = pulumiTaggedImages
```

This gets all of the docker images specified in the default values.yaml file of the tezos chart. So no need to hardcode them in pulumi.

Reply:
I've incorporated this suggestion.